syft/pkg/gguf.go

```diff
@@ -0,0 +1,47 @@
+package pkg
+
+// GGUFFileHeader represents metadata extracted from a GGUF (GPT-Generated Unified Format) model file.
+// GGUF is a binary file format used for storing model weights for the GGML library, designed for fast
+// loading and saving of models, particularly quantized large language models.
+type GGUFFileHeader struct {
+	// ModelFormat is always "gguf"
+	ModelFormat string `json:"modelFormat" cyclonedx:"modelFormat"`
+
+	// ModelName is the name of the model (from general.name or filename)
+	ModelName string `json:"modelName" cyclonedx:"modelName"`
+
+	// ModelVersion is the version of the model (if available in header, else "unknown")
+	ModelVersion string `json:"modelVersion,omitempty" cyclonedx:"modelVersion"`
+
+	// FileSize is the size of the GGUF file in bytes (best-effort if available from resolver)
+	FileSize int64 `json:"fileSize,omitempty" cyclonedx:"fileSize"`
+
+	// Hash is a content hash of the metadata (for stable global identifiers across remotes)
+	Hash string `json:"hash,omitempty" cyclonedx:"hash"`
```

Comment:
we should beef up teh description for this to explain that this is a hash of all k-v pairs. The name should probably change to convey this too (maybe). And this should be colocated in the struct to the Header field